### ISSUE TITLE
Blank Page after Install

### ISSUE THREAD

**Issue by sai-mike** [Offerel/roundcube_primitivenotes#1]
I used a git clone to install to my roundcube plugins directory and then renamed the directory as 'primitivenotes'.  I created a base path on my server, copied config.inc.php.dist to config.inc.php and edited it to include the base path I created.  I also set read/write permissions on the base path directory for the www-data group.

When I login, I get the Notes icon, and when I click that I get a page where the icons partially overwrite my logo.  If I click on New Note, I get a page refresh and no content (see attached). I clicked on the Upload icon, selected a jpeg file, got a page refresh, and no content.  I checked my base path and no new user folder is being created there.
![notespage](https://user-images.githubusercontent.com/35618397/35179400-84a3ca6e-fd4e-11e7-8ba8-0e19f808431f.jpg)

**Comment by Offerel** [comment]
I don't think this is a problem with the permissions. But maybe a JavaScript issue or some missing files or references. Can you please tell me if there are some errors in your browsers webconsole? In Firefox you can reach this with pressing F12 on Windows and Linux. Chrome should be the same if I remember correctly. In Internet Explorer I'm not sure.

**Comment by Offerel** [comment]
I don't think this is a problem with the permissions. But maybe a JavaScript issue or some missing files or references. Can you please tell me if there are some errors in your browsers webconsole? In Firefox you can reach this with pressing F12 on Windows and Linux. Chrome should be the same if I remember correctly. In Internet Explorer I'm not sure.

**Comment by Offerel** [comment]
I don't think this is a problem with the permissions. But maybe a JavaScript issue or some missing files or references. Can you please tell me if there are some errors in your browsers webconsole? In Firefox you can reach this with pressing F12 on Windows and Linux. Chrome should be the same if I remember correctly. In Internet Explorer I'm not sure.

**Comment by Offerel** [comment]
Regarding the overlapping logo, i can try to move the toolbar a little bit to the right side. Currently it's positioned xxx pixels from the left browser border. I try to find, if I can place it in context to the logo wide.

**Comment by Offerel** [comment]
I have now installed Chrome and changed some css parts for this, since chrome seems not really accurate with css styles. Additionally, i have moved the toolbar in classic skin more to the right.

I have also tried to install the plugin directly from a Github checkout, to see if i miss some files. But i havent found nothing. The plugin works here, also with this checkout. But maybe there was a problem, creating more than one subfolder. I have fixed this also. I dont know if this was the problem. However, maybe you can try this new version.

**Comment by sai-mike** [comment]
I will pull the new version now. I run Firefox so I started webconsole and found this on the Debugger tab:

Could not load the source for https://<my_server>/roundcube/plugins/primitivenotes/notes.php.
[Exception... "Component returned failure code: 0x80470002 (NS_BASE_STREAM_CLOSED) [nsIInputStream.available]"  nsresult: "0x80470002 (NS_BASE_STREAM_CLOSED)"  location: "JS frame :: resource://devtools/shared/base-loader.js -> resource://devtools/shared/DevToolsUtils.js :: onResponse :: line 482"  data: no]
Stack: onResponse@resource://devtools/shared/base-loader.js -> resource://devtools/shared/DevToolsUtils.js:482:23
onStopRequest@resource://gre/modules/NetUtil.jsm:131:17
Line: 482, column: 0

**Comment by sai-mike** [comment]
The icons are now far to the right of the logo, but I still have the same error in the webconsole debugger. I think I found better info in the Apache error log:

[Sat Jan 20 09:59:25.111030 2018] [:error] [pid 22597] [client 192.168.xx.28:50083] PHP Parse error:  syntax error, unexpected '>' in /var/www/roundcube/plugins/primitivenotes/notes.php on line 114, referer: https://<my_server>/roundcube/?_task=notes

**Comment by Offerel** [comment]
ahh that was really helpful. that is the sort function. it sorts the notes by lastmodified timestamp. This seems to work differently on your server. I have tried to fix it. Please try the new commit.

**Comment by Offerel** [comment]
Ahh thats better now. Since im working also most time with Firefox, i cant reproduce the problem with the height of the plugin div. Since i changed the css heavily the at last commits, maybe this could a caching problem. Can you please reload the page in a private firefox window? This should load the page without any caching and in a completely new session.

JPG and PDF display is a other problem, i have to fix. Only HTML and TXT (and Markdown, which is broken currently) data are displayed directly. Other formats are parsed as base64 and presented in a html object tag. I assume, that i didn't read the mimetype correctly.  I try to fix this tomorrow.

**Comment by Offerel** [comment]
changed now the mimetype detection. please dont use markdown files with this commit, since there is now a experimental ckeditor plugin insert. but currently ckeditor removes the linebreaks, which breaks also markdown files.

**Comment by sai-mike** [comment]
Opening in a private window didn't change my display problem.  PDFs and JPEGs are now displaying, but I only get a slice of the pdf/jpeg in a window with a second scrollbar now.
![notes6](https://user-images.githubusercontent.com/35618397/35189362-cc331d6c-fdfd-11e7-926d-0b253539327b.jpg)

**Comment by Offerel** [comment]
I see. This seems to be a new issue after changing the mimetype problem. I fear it was late yesterday... Sorry for that. I fix it.

**Comment by Offerel** [comment]
I have tried a new commit. The div uses now at least on my Firefox the full height. Partial markdown support is now also available.

**Comment by sai-mike** [comment]
I did a pull - no change in results.  I have tested on a Windows Notebook & Desktop running Firefox, a Linux Mini-Notebook running Firefox, my iPad running Safari, and the Windows Desktop running IE 11.  All tests show the same short vertical height.  IE 11 is also showing a short width:

![notes7](https://user-images.githubusercontent.com/35618397/35199261-b17927ec-feb0-11e7-9361-a42326f67ed4.png)

**Comment by Offerel** [comment]
That's really strange. Do you have changed some other css styles somewhere? Maybe with some other plugin? Can you please try to deactivate other plugins and use temporary only the notes plugin? 

Please use the private window for a test with deactivated other plugins, to exclude caching issues.

**Comment by sai-mike** [comment]
Disabling my other plugins didn't change anything.  So I re-enabled them and on a hunch, I switched to the Larry skin.  Problem is fixed using Larry (except the icons still run into my logo which wasn't sized for Larry).  Switching back to Classic, the problem returns:

![notes8](https://user-images.githubusercontent.com/35618397/35201946-178b1cca-fed4-11e7-9013-e85957c50fed.png)

**Comment by Offerel** [comment]
In the commit this morning, i have moved the css in on file for both skins. Both are now identical. Please try this new variant.

**Comment by Offerel** [comment]
Now I really don't have any ideas anymore. I even reinstalled Roundcube and installed PrimitiveNotes as the only plugin. Both Larry and Classic display the full height. But also the IE and Chrome display the plugin correctly with both skins. In the version on Github, various specifications in the CSS are currently calculated via variables. I'm sorry to say that Safari and Internet Explorer can't handle it. In my local version I replaced this with conventional values, so that at least the IE works. Unfortunately I can't test Safari. 

At the moment I can't commit this version, because I am working on other changes at the moment. Once this is complete, there will be a new commit. Maybe your problem is solved with it. But I can't imagine that, because Firefox and Chrome didn't have any problems with the variables so far.

**Comment by Offerel** [comment]
Ok, there is something new. I have added a new branch "ajax_test". If have rewritten from scratch most of the requests. With this, its no more needed to load the complete plugin every time. Locally this is a lot faster on my Raspberry Pi. With this, some css files have changed also. It's not minimized yet at all, since its not complete. But nearby.

Maybe you can try this new branch. I would recommend to rename the old plugin dir before and checkout this new branch. 

I have tested this new branch with Firefox, Chrome and IE11. It should work at least with these Browsers. I hope your problem is now fixed also.

**Comment by sai-mike** [comment]
I backed up my master branch plugin directory and checked out the ajax_test branch.  I then opened Roundcube in both a Firefox Private Window and in IE 11 on my notebook.  With the Classic Skin, both browsers had the same vertical height problem as Firefox had before.  With the Larry Skin, both browsers were working properly.

I have a spare Raspberry PI 3 sitting on the shelf so I am going to load a fresh install of Apache & Roundcube on that box and see what happens.  It will probably take me a day or two to get that test done.

**Comment by Offerel** [comment]
Currently im on Roundcube 1.3.4 if that matters. But im sure, it works wither older versions to. Ahh and as the webserver, i use Nginx with PHP7.0-fpm. But this should also have no impact on the Layout. The only thing, which can have an impact in my opinion, can be the CSP header policys or another plugin or a modified CSS in Rondcube itself.

**Comment by sai-mike** [comment]
I pulled my RPi3 off the shelf, pulled the SD Card, formatted it, and loaded the latest version of Raspbian Stretch.

I then installed MariaDB, Apache 2.4, PHP 7.0, and Roundcube 1.3.4.  Roundcube only has the default plugins enabled - archive and zipdownload.  I tested the connection to my mail server and made sure Roundcube was working properly.

I then did a git clone of the master branch of primitivenotes, setup the notes storage directory, set up permissions, etc.

I ran Roundcube both from the RPi using localhost in Chromium and from my Windows Desktop w/ Firefox connecting to the RPi.  I get exactly the same results as when running Roundcube on my Debian Mail Server - the Larry Skin displays primitivenotes correctly and the Classic Skin has the same vertical height problem.

![notes](https://user-images.githubusercontent.com/35618397/35413017-fa53b912-01d2-11e8-983b-59adb253e169.png)

**Comment by Offerel** [comment]
That are bad news... and i have no solution for you. I see some AddOns in your Firefox toolbar. Can you please temporary disable all AddOns and reload the page in a private window afterwards? Its only something like a last hope. Since you have the same problem with other browsers to, but its worth a test...

Since the css is the same file for both skins, i have no idea whats going on. I have tested this now again on 5 computers with IE and Firefox and all of them have no problem so far. It would be great to hear from other users, if they have the same problems, but i have no response yet from others regarding the layout.

**Comment by sai-mike** [comment]
I verified there is a problem saving manually entered notes in the master branch.  Uploads are saved properly. I uploaded a text file and that was saved.  However, when I clicked on New and entered a note it was not saved.  No error in roundcube logs.  No error in the apache logs.

I will checkout the ajax_test branch and try that.

**Comment by Offerel** [comment]
in ajax it should work, i know that its broken in current master. If i know where the problem with the classic skin comes from, i will merge ajax to master.

**Comment by sai-mike** [comment]
I checked out ajax_test and that didn't fix the vertical height issue with Classic.  I also noticed that I can't create a New note with this branch - when I click the New icon I get a screen flash and nothing else.  If I try to edit one of my prior uploads, I can change the note title but clicking the Save icon doesn't save the change. I spot checked the notes.php and README.md files to make sure I received the ajax_test versions and I did.

**Comment by Offerel** [comment]
I fixed the bug with the new note button. Since i fixed it only with the editor on my smartphone, i have also seen that some old files are loaded from the cache. After cleaning the cache, the "new button" creates now a new note again and also saving existing notes do work. Saving itself is not changed.

There are changes with editing notes in this version. You can now edit all supported formats. But only editing html, txt and markdown means, that you can edit the note content. If you edit pdf, jpg or png, you can change with edit only the title and the keywords.

Format for a new note is always html.

The config has changed also. There is now a new parameter for the editor. You can use with this, also the internal tinymce editor from Roundcube. Tinymce configuration isn't complete yet but it works basically. For example the save and other buttons is missing and some function's do not work as expected. 

Currently you must explicitly choose one or the other, later ckeditor will be the default. I would recommend to set this to ckeditor for now.

**Comment by Offerel** [comment]
Heureka! Maybe i have found something regarding the classic skin issue. Could you please try the latest commit to the ajax branch from this morning? I keep the fingers crossed, in hope that we have now a solution... 

BTW: The editor defaults now to ckeditor, also if the parameter is empty or not specified.

**Comment by sai-mike** [comment]
You got it - fixed!  I will let you merge all changes then download and test the edit issues I was seeing before.

**Comment by Offerel** [comment]
Ahh that sounds good. It was an old relict from my CalDAVZap plugin. I really hope, that this was the only one. I close this issue now, merge the branch and make a new release. Since i can't locally find a edit issue, i think that there is not such one. But when im wrong and you have a issue with saving, please open a new issue for that.

And thank you so much for your patience.
